ewwep.py

```diff
@@ -2135,38 +2137,50 @@ def apply_combat_mods(user_data = None, desired_type = None, target = None):
 		if enemy_data.enemytype == ewcfg.enemy_type_doubleheadlessdoublehorseman:
-			defeat_response = "***AHA... AHAHAHAHA...***\n*COUGH*... *HACK*...\nYOU HAVE ALL TRULY PUT ON A SPLENDID PERFORMANCE.\nI KNOW WHEN I AM DEFEATED. PLEASE, TAKE THIS MEDALLION...\nIT IS A TOKEN OF MY EXTREME PRAISE.\nFOR NOW THOUGH, THIS IS WHERE I GET OFF.\nSAVE A SEAT FOR ME, WON'T YOU, PHOEBUS?\n"
-			resp_cont.add_channel_response(cmd.message.channel.name, defeat_response)
-
-			# Give the medallion to everyone in the underworld
-			# The horseman only spawns in the underworld and can't leave, so using current district_data is fine
-			life_states = [ewcfg.life_state_juvenile, ewcfg.life_state_enlisted, ewcfg.life_state_executive]
-			all_users_in_underworld = district_data.get_players_in_district(life_states=life_states)
-
-			for user in all_users_in_underworld:
-				underworld_user_data = EwUser(id_user=user, id_server=user_data.id_server)
-				underworld_player_data = EwPlayer(id_user=user)
-
-				medallion = ewcfg.medallion_results[0]
-				medallion_props = ewitem.gen_item_props(medallion)
-
-				medallion_id = ewitem.item_create(
-					item_type=medallion.item_type,
-					id_user=underworld_user_data.id_user,
-					id_server=underworld_user_data.id_server,
-					item_props=medallion_props
-				)
-
-				# Soulbind the medallion. A player can get multiple medallions, but later on a new command could be added to destroy them.
-				# I imagine this would be something similar to how players can destroy Australium Wrenches in TF2, which broadcasts a message to everyone in the game, or something.
-				ewitem.soulbind(medallion_id)
+			market_data = EwMarket(id_server=cmd.message.server.id)
+			horseman_deaths = market_data.horseman_deaths
+			
+			if horseman_deaths == 0:
+				defeat_response = "***AHA... AHAHAHAHA...***\n*COUGH*... *HACK*...\nYOU HAVE ALL TRULY PUT ON A SPLENDID PERFORMANCE.\nI KNOW WHEN I AM DEFEATED. PLEASE, TAKE THESE GIFTS OFF OF MY CORPSE...\nTHEY ARE OF NO USE TO ME ANYMORE.\nFOR NOW THOUGH, THIS IS WHERE I GET OFF.\nSAVE A SEAT FOR ME, WON'T YOU, PHOEBUS?\n"
+			else:
+				defeat_response = "***GAHAHAH... AHAHA...***\n*WHEEZE*... *PUKE*...\nBESTED A SECOND TIME...\nIF I WEREN'T GONE FOR GOOD... THIS WOULD FEEL LIKE AN INSULT.\nNOW, HOWEVER, I HAVE REACHED MY LIMIT.\nTHE GREAT BEYOND CALLS TO ME ONCE AGAIN, AND I ANSWER.\nI JUST HOPE I HAVE THE HEART TO TELL HIM HOW I FEEL...\nUNTIL WE MEET AGAIN AT NEXT DOUBLE HOLLOW'S EVE, ***MORTALS!!!***\n"
 
-				medallion_player_response = "**{} has been gifted the Double Halloween Medallion!!**\n".format(underworld_player_data.display_name)
-				resp_cont.add_channel_response(cmd.message.channel.name, medallion_player_response)
+			resp_cont.add_channel_response(cmd.message.channel.name, defeat_response)
+			
+			
+			market_data.horseman_deaths += 1
+			market_data.horseman_timeofdeath = int(time_now)
+			market_data.persist()
+
+			# # Give the medallion to everyone in the underworld
```

Comment:
why are we now doing this manually instead of automatically?

Reply:
It’s come to my attention that a fair amount of people are actually going to be playing supportive roles rather than dealing damage. If the horseman is to have limited spawns, I want to reward those who, while they may have died, still put in a considerable amount of effort, given that they won’t have that many chances to get it in the first place.